src/main/generator/namespace.ts

```diff
@@ -0,0 +1,174 @@
+import {
+    ConstDefinition,
+    EnumDefinition,
+    ExceptionDefinition,
+    ServiceDefinition,
+    StructDefinition,
+    TypedefDefinition,
+    UnionDefinition,
+} from '@creditkarma/thrift-parser'
+import { dirname, resolve } from 'path'
+import * as ts from 'typescript'
+import ResolverFile from '../resolver/file'
+import ResolverNamespace from '../resolver/namespace'
+import { IMakeOptions, IRenderedFile, IRenderer, IStatementMap } from '../types'
+
+export default class NamespaceGenerator {
+    public types: IStatementMap = {}
+    public renderer: IRenderer
+    public namespace: ResolverNamespace
+    public options: IMakeOptions
+    public outDir: string
+
+    constructor(
+        renderer: IRenderer,
+        namespace: ResolverNamespace,
+        outDir: string,
+        options: IMakeOptions,
+    ) {
+        this.renderer = renderer
+        this.namespace = namespace
+        this.outDir = outDir
+        this.options = options
+    }
+
+    public addStatements(
+        type: string,
+        statements: Array<ts.Statement>,
+        namespaced: boolean,
+    ) {
+        this.types[type] = {
+            namespaced,
+            statements,
+        }
+    }
+    public renderFileForType(
+        typeName: string,
+        statements: Array<ts.Statement>,
+    ): IRenderedFile {
+        const path = resolve(dirname(this.namespace.path), `${typeName}.ts`)
+
+        return {
+            outPath: path,
+            namespace: this.namespace,
+            statements: [
+                ...this.renderer.renderIncludes(
+                    this.namespace,
+                    this.options,
+                    '__NAMESPACE__',
```

Comment:
This might not be the best name to use here, but works well enough for now.  Eventually, I would prefer to import just the referenced types from the current namespace (there shouldn't be any collisions when doing this).  But this PR is already really complicated and I didn't want to add that logic in the same PR